src/providers/wms/qgswmsprovider.cpp

```diff
@@ -1018,7 +1021,19 @@ bool QgsWmsProvider::readBlock( int bandNo, QgsRectangle  const &viewExtent, int
   if ( ptr )
   {
     // If image is too large, ptr can be NULL
-    memcpy( block, ptr, myExpectedSize );
+    if ( mConverter )
+    {
+      std::vector<float> data;
+      data.resize( myImageSize );
+      for ( int w = 0; w < pixelWidth; w++ )
+        for ( int h = 0; h < pixelHeight; h++ )
+          mConverter->convert( image->pixel( w, h ), &data[h * pixelWidth + w] );
```

Comment:
I have a vague recollection that calling `QImage::pixel()` can slow things down - it may be worth having an optimized case - if the QImage format is Format_RGB32 or Format_ARGB32, then we could get raw pointer to data `QRgb* ptr = image->bits();` and avoid the extra per-pixel calls (and also the for loop could be just i=0 to w*h: `mConverter->convert(*ptr++, outPtr++)`